samples/AzureSDK/AzureEdgeOta/AzureEdgeOTAEngine/Program.cs

```diff
@@ -0,0 +1,353 @@
+            }
+        }
+    }
+    // And update the reported twin
+    string status = isRunning ? CodeRunning : CodeNotRunning;
+    reported.Add(CodeRunning, status);
+    azure.UpdateReportedProperties(reported);
+}
+
+void TwinUpdated(object sender, TwinUpdateEventArgs e)
+{
+    ProcessTwinAndDownloadFiles(e.Twin);
+    twinUpated?.Invoke(null, new object[] { e.Twin });
+}
+
+void LoadAssemblies()
+{
+    // Now load all the assemblies we have on the storage
+    var files = Directory.GetFiles(RootPath);
+    foreach (var file in files)
+    {
+        if (file.EndsWith(".pe"))
+        {
+            using FileStream fspe = new FileStream(file, FileMode.Open, FileAccess.Read);
+            Trace($"{file}: {fspe.Length}");
+            var buff = new byte[fspe.Length];
+            fspe.Read(buff, 0, buff.Length);
+            // Needed as so far, there seems to be an issue when loading them too fast
+            fspe.Close();
+            fspe.Dispose();
+            Thread.Sleep(20);
+            bool integrity = true;
+            string strsha = string.Empty;
+            // Check integrity if we just downloaded it
+            if (filesToDownload != null)
+            {
+                integrity = false;
+                var sha256 = SHA256.Create().ComputeHash(buff);
+                strsha = BitConverter.ToString(sha256);
+                var fileName = file.Substring(file.LastIndexOf('\\') + 1);
+                foreach (FileSettings filesetting in filesToDownload)
+                {
+                    if (filesetting.FileName.Substring(filesetting.FileName.LastIndexOf('/') + 1) == fileName)
+                    {
+                        if (strsha == filesetting.Signature)
+                        {
+                            integrity = true;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            if (!integrity)
+            {
+                Trace("Error with file signature");
+                TwinCollection reported = new();
+                reported.Add(CodeRunning, $"{IntegrityError}:{file} - {strsha}");
+                azure.UpdateReportedProperties(reported);
+                break;
+            }
+
+            var ass = Assembly.Load(buff);
+            var typeToRun = ass.GetType(OtaRunnerName);
+            if (typeToRun != null)
+            {
+                toRun = ass;
+            }
+        }
+    }
+}
+
+void GoToSleep()
+{
+    Trace($"Set wakeup by timer for {millisecondsToSleep} minutes to retry.");
+    Sleep.EnableWakeupByTimer(new TimeSpan(0, 0, millisecondsToSleep, 0));
+    Trace("Deep sleep now");
+    Sleep.StartDeepSleep();
+}
+
+void DownloadBinaryFile(string url, string sas)
+{
+    int bytesRead = 0;
+    int totalBytes = 0;
+    HttpWebResponse httpWebResponse;
+    string fileName = url.Substring(url.LastIndexOf('/') + 1);
+
+    var httpWebRequest = (HttpWebRequest)WebRequest.Create($"{url}?{sas}");
```

Comment:
You should use the new HttpClient instead, it's sooo much cleaner.

Reply:
right, I started this code before :-D So will adjust then